control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go

```diff
@@ -0,0 +1,209 @@
+	p := Params{
+		Images: Images{
+			NodeTuningOperator: images["cluster-node-tuning-operator"],
+			NodeTunedContainer: images["cluster-node-tuning-operator"],
+		},
+		ReleaseVersion: version,
+	}
+
+	p.DeploymentConfig.Scheduling.PriorityClass = config.DefaultPriorityClass
+	p.DeploymentConfig.SetDefaults(hcp, nil, utilpointer.IntPtr(1))
+	p.DeploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
+
+	p.DeploymentConfig.SetDefaultSecurityContext = setDefaultSecurityContext
+	p.HostedClusterName = hcp.Name
+
+	return p
+}
+
+func ReconcileClusterNodeTuningOperatorMetricsService(svc *corev1.Service) error {
+	svc.Annotations = map[string]string{
+		"service.beta.openshift.io/serving-cert-secret-name": "node-tuning-operator-tls",
+	}
+
+	svc.Labels = map[string]string{
+		"name": metricsServiceName,
+		"hypershift.openshift.io/control-plane-component": "node-tuning-operator",
+	}
+	svc.Spec.Ports = []corev1.ServicePort{
+		{
+			Name:       "metrics",
+			Port:       60000,
+			Protocol:   corev1.ProtocolTCP,
+			TargetPort: intstr.FromInt(60000),
+		},
+	}
+	svc.Spec.Selector = map[string]string{
+		"name": metricsServiceName,
+	}
+
+	return nil
+}
+
+func ReconcileClusterNodeTuningOperatorServiceMonitor(sm *prometheusoperatorv1.ServiceMonitor, clusterID string, metricsSet metrics.MetricsSet) error {
+	sm.Spec.Selector.MatchLabels = map[string]string{
+		"name": metricsServiceName,
+		"hypershift.openshift.io/control-plane-component": metricsServiceName,
+	}
+	sm.Spec.NamespaceSelector = prometheusoperatorv1.NamespaceSelector{
+		MatchNames: []string{sm.Namespace},
+	}
+	targetPort := intstr.FromInt(60000)
+	sm.Spec.Endpoints = []prometheusoperatorv1.Endpoint{
+		{
+			Interval:   "15s",
+			TargetPort: &targetPort,
+			Scheme:     "https",
+			Path:       "/metrics",
+			TLSConfig: &prometheusoperatorv1.TLSConfig{
+				SafeTLSConfig: prometheusoperatorv1.SafeTLSConfig{
+					ServerName: metricsServiceName + "." + sm.Namespace + ".svc",
+					Cert: prometheusoperatorv1.SecretOrConfigMap{
+						Secret: &corev1.SecretKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "node-tuning-operator-tls",
+							},
+							Key: "tls.crt",
+						},
+					},
+					KeySecret: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: "node-tuning-operator-tls",
+						},
+						Key: "tls.key",
+					},
+					CA: prometheusoperatorv1.SecretOrConfigMap{
+						ConfigMap: &corev1.ConfigMapKeySelector{
+							LocalObjectReference: corev1.LocalObjectReference{
+								Name: "openshift-service-ca.crt",
+							},
+							Key: "service-ca.crt",
+						},
+					},
+				},
+			},
+			MetricRelabelConfigs: metrics.NTORelabelConfigs(metricsSet),
+		},
+	}
+
+	util.ApplyClusterIDLabel(&sm.Spec.Endpoints[0], clusterID)
+
+	return nil
+}
+
+func ReconcileDeployment(dep *appsv1.Deployment, params Params) error {
+	dep.Spec.Replicas = utilpointer.Int32(1)
```

Comment:
nit: don't need replicas, is set by setDefaults()

Reply:
Thanks, I removed this line.